cmd/root.go

```diff
@@ -0,0 +1,123 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	models "github.com/kyallanum/athena/v1.0.0/models"
+	config "github.com/kyallanum/athena/v1.0.0/models/config"
+	library "github.com/kyallanum/athena/v1.0.0/models/library"
+	"github.com/kyallanum/athena/v1.0.0/utils"
+	"github.com/spf13/cobra"
+)
+
+var configFile, logFile string
+
+var rootCmd = &cobra.Command{
+	Use:          "athena [flags]",
+	Short:        "A text and log file parser to discern important information",
+	SilenceUsage: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(configFile) == 0 {
+			return fmt.Errorf("")
+		}
+		if len(logFile) == 0 {
+			return fmt.Errorf("")
+		}
+		fmt.Println("Athena v1.0.0 Starting")
+
+		fmt.Println("Getting Configuration File: ", configFile, "...")
+		configuration, err := utils.CreateConfiguration(configFile)
+		if err != nil {
+			return err
+		}
+		fmt.Println("Configuration Loaded")
+
+		fmt.Println("Loading Log File: ", logFile, "... ")
+		logFileContents, err := utils.LoadLogFile(logFile)
+		if err != nil {
+			return err
+		}
+		fmt.Println("Log File Loaded")
+
+		library, err := resolveLogFile(logFileContents, configuration)
+		if err != nil {
+			return err
+		}
+
+		if err = printSummary(library); err != nil {
+			return err
+		}
+		return nil
+	},
```

Comment:
I'm not quite sure if I like the idea of putting the whole original main.go process under RunE. I don't believe the object controlling the command line argument should run the whole process, but rather only take care of the command line argument. Instead, I believe the majority of the process execution should be under ``func Execute()`` however I'm open to discussion on this.

Reply:
As an extra comment on this, rootCmd is a global variable, so having the full function execution coming from a global variable is not preferred. I believe I would prefer that it just assign the values from the command line arguments to global variables themselves, and then those be used under the ``Execute()`` function.

Reply:
> I'm not quite sure if I like the idea of putting the whole original main.go process under RunE. 

In general, applications using the cobra package are usually structured in this way [(refer to the user guide here)](https://github.com/spf13/cobra/blob/main/site/content/user_guide.md).

Reply:
If in case you'd prefer running all your logic in main.go, we could return a pointer to the configFile and logFile variables to main via the cmd.Execute function.

Reply:
I don't think we should be running logic in main.go. I believe we should be running logic directly in the Execute() function. Although it shows that you can put logic in the "RunE" portion of the command, it doesn't say that all of the program logic should go inside of the command. What I think would be okay, is if we further abstract the program logic to another function, specifically in the case that in the future, we want to add new commands in the future.

Reply:
Okay, so what if we move the functions `printSummary` and `resolveLogFile` to the utils module as they are both helper functions? Then we can move all the logic out of `RunE` that isnt concerned with arg parsing into `Execute`.

Reply:
@abh1sheke I'm actually working on removing the utils package right now. See: https://dave.cheney.net/2019/01/08/avoid-package-names-like-base-util-or-common

You can check out my branch: update/code-cleanup and see what changes I am making. Its not the most of difference.

That being said, with this layout, I believe ``resolveLogFile`` would go in the ``models/logfile/logfile.go``, meanwhile I'm not sure what I'm doing with the Summary.go file, so we can put it there for now, and just make sure it all works.

Reply:
Right now there's an import cycle conflict between modules `util` and `models` so moving `resolveLogFile` into models isn't really possible without further further changes.

Reply:
Right at the main branch there's an import cycle conflict between `util` and `models` , so moving `resolveLogFile` into models wouldn't work. What if you merge this request into main and then rebase branch `update/code-cleanup`?

Reply:
@abh1sheke I'm working on some finishing touches on ``update/code-cleanup`` and then once I do, I will merge that into main. This will then completely remove utils, and there won't be an import cycle anymore. I think that's probably our best bet into getting this merged in properly. Thanks for your patience on this.

Reply:
No issues.  Shall await changes to main :)

Reply:
@abh1sheke okay you should be good to go.

Reply:
@kyallanum on it, but there's still an import cycle conflict if in case `resolveLogFile` is moved into `models/logs/logfile` as `models/config` depends on logs. any inputs on that? or do we just keep it in main?

Reply:
@abh1sheke let's just keep both in the cobra command for the moment and I'll see about moving it around later. Feel free to open up a new PR, and I'll review all changes made and then go from there.